src/components/Popover/src/Popover.vue

```diff
@@ -0,0 +1,278 @@
+		>
+			<slot name="content" />
+		</portal>
+	</div>
+</template>
+
+<script>
+import { throwError } from '@square/maker/utils/debug';
+import assert from '@square/maker/utils/assert';
+import { Portal } from '@linusborg/vue-simple-portal';
+import { v4 as uuid } from 'uuid';
+import { PopoverConfigKey, PopoverAPIKey } from './keys';
+
+const MAX_TETHER_VNODE = 1;
+
+const getMinWidth = (minWidth, tetherMinWidth, reference) => {
+	if (!tetherMinWidth) {
+		return minWidth;
+	}
+
+	return minWidth || reference.offsetWidth;
+};
+
+const createPopperConfig = ({
+	placement,
+	offset,
+	tetherMinWidth,
+	minWidth,
+}) => ({
+	placement,
+	modifiers: [
+		{
+			name: 'offset',
+			options: { offset },
+		},
+		{
+			name: 'preventOverflow',
+			options: {
+				padding: 24,
+				altBoundary: true,
+			},
+		},
+		{
+			name: 'minWidth',
+			enabled: tetherMinWidth || !!minWidth,
+			phase: 'beforeWrite',
+			requires: ['computeStyles'],
+			fn({ state: { styles, rects } }) {
+				const popoverMinWidth = getMinWidth(minWidth, tetherMinWidth, rects.reference);
+				styles.popper.minWidth = `${popoverMinWidth}px`;
+			},
+			effect({ state: { elements } }) {
+				const popoverMinWidth = getMinWidth(minWidth, tetherMinWidth, elements.reference);
+				elements.popper.style.minWidth = `${popoverMinWidth}px`;
+			},
+		},
+	],
+});
+
+export default {
+	name: 'Popover',
+
+	components: {
+		Portal,
+	},
+
+	inject: {
+		popoverApi: {
+			from: PopoverAPIKey,
+			default: () => throwError('No popover layer detected', 'Popover'),
+		},
+
+		popoverConfig: {
+			from: PopoverConfigKey,
+			default: {
+				config: undefined,
+			},
+		},
+	},
+
+	props: {
+		/**
+		 * Starting position of the popover. Not guaranteed if too close to overflow.
+		 */
+		placement: {
+			type: String,
+			default: 'bottom-start',
+			validator: (placement) => [
+				'auto', 'auto-start', 'auto-end',
+				'top', 'top-start', 'top-end',
+				'right', 'right-start', 'right-end',
+				'bottom', 'bottom-start', 'bottom-end',
+				'left', 'left-start', 'left-end',
+			].includes(placement),
+		},
+
+		/**
+		 * Distance from tether element
+		 */
+		distanceOffset: {
+			type: Number,
+			default: 8,
+		},
+
+		/**
+		 * Offset from base position (Y for left/right placement, X for top/bottom placement)
+		 */
+		skiddingOffset: {
+			type: Number,
+			default: 0,
+		},
+
+		/**
+		 * Absolute min width of popover, overrides tetherMinWidth
+		 */
+		minWidth: {
+			type: Number,
+			default: 0,
+		},
+
+		/**
+		 * Set min width of popover to tether width, overridden by minWidth
+		 */
+		tetherMinWidth: {
+			type: Boolean,
+			default: true,
+		},
+
+		/**
+		 * Configures the popover to hide itself if the tether element goes out of view
+		 */
+		shouldHideOnOverflow: {
+			type: Boolean,
+			default: true,
+		},
+
+		/**
+		 * ID of popper
+		 */
+		id: {
+			type: String,
+			default: () => uuid(),
+		},
+
+		/**
+		 * Custom popover config
+		 */
+		customConfig: {
```

Comment:
I'm hesitant to reveal a third-party configuration option just yet until we have more scenarios that are accounted for in our base props. Having multiple props that configure the same type of options can cause issues as you have to make internal choices of which one to use if multiple are set.

Reply:
Unfortunately we need this for mini cart, especially if you're having me drop the other props. Otherwise we'll have to roll our own solution for it.

Reply:
Yup so what I've done is remove all that and just exposed a `modifiers` prop that takes in an array.